ext/js/language/translator.js

```diff
@@ -457,7 +457,7 @@ export class Translator {
         /** @type {import('translation-internal').DatabaseDeinflection[]} */
         const deinflections = [];
         const used = new Set();
-        /** @type {Map<string, Map<string, Map<string, string>>>} */
+        /** @type {Map<string, Map<any, Map<any, string>>>} */
```

Comment:
I don't really get all the nuances of the eslint/type rules, but I'm surprised it's letting an explicit `any` pass. I'd expect `SafeAny` at least to be mandatory. I think `unknown` is better here to match the rest of the types. 

Also, I'd move this type definition for "TextCache" to `translation-internal.d.ts`, and also the `Map<string, unknown>` for "ProcessorVariant", since they are not primitive and are used in multiple places.